addons/point_of_sale/static/src/js/gui.js

```diff
@@ -184,20 +184,29 @@ var Gui = core.Class.extend({
     close_other_tabs: function() {
         var self = this;
 
+        // avoid IE to close itself
+        var now = +new Date();
+
         localStorage['message'] = '';
         localStorage['message'] = JSON.stringify({
             'message':'close_tabs',
             'session': this.pos.pos_session.id,
+            'time': now,
         });
 
+        // storage events are (theorically) triggered only when the
+        // localstorage is updated in a different tab
+        // IE does trigger an event in the same tab
+        // https://connect.microsoft.com/IE/feedback/details/774798/localstorage-event-fired-in-source-window
```

Comment:
As an aside I briefly skimmed the linked W3C recommendation [1] and I'm not sure we can blame IE in fact. There is a *Note* mentioning:

> When the setItem(), removeItem(), and clear() methods are invoked, events are fired on the Window objects of other Documents that can access the newly stored or removed data, as defined in the sections on the sessionStorage and localStorage attributes.

This doesn't prohibit the storage event from being fired on the same Window and I'm not sure this *Note* is a formal part of some W3C specification. Furthermore later on in section 4.4 it mentions:

> The storage event is fired on a Document's Window object when a storage area changes, as described in the previous two sections (for session storage, for local storage).

Which doesn't make any kind of distinction between the current window or other windows. So I don't agree with the bug reporter that concludes this is a browser bug.

[1] https://www.w3.org/TR/webstorage/#storage